### PR TITLE
Update to react native module 3.1.1

### PR DESCRIPTION
**Pull request by iujames**
also: example app update to react native 0.72.4 -- this is responsible for most of the changes in this PR. This update in the example is required to use Android API 34.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -42,7 +42,7 @@
   },
   "peerDependencies": {
     "@segment/analytics-react-native": "*",
-    "@appcues/react-native": "^3.1.0"
+    "@appcues/react-native": "^3.1.1"
```
Comment (iujames) [inline package.json:45]: ignoring the example app updates - this is the actual version bump for the plugin

File: example/package.json [hunk 2]
```diff
@@ -9,26 +9,31 @@
     "start": "react-native start"
   },
   "dependencies": {
-    "@appcues/react-native": "^3.1.0",
+    "@appcues/react-native": "^3.1.1",
```
Comment (iujames) [inline example/package.json:12]: and this is the version bump in the example app

File: example/android/gradle/wrapper/gradle-wrapper.properties [hunk 3]
```diff
@@ -1,5 +1,6 @@
 distributionBase=GRADLE_USER_HOME
 distributionPath=wrapper/dists
-distributionUrl=https\://services.gradle.org/distributions/gradle-7.5-bin.zip
+distributionUrl=https\://services.gradle.org/distributions/gradle-8.0.1-all.zip
+networkTimeout=10000
```
Comment (andretortolano) [inline example/android/gradle/wrapper/gradle-wrapper.properties:4]: what is this property?

**Review by andretortolano — APPROVED** [review]

File: example/android/gradle/wrapper/gradle-wrapper.properties (hunk 3 above)
Comment (iujames) [reply example/android/gradle/wrapper/gradle-wrapper.properties:4]: good question!? I'm not sure. This all came over from the RN update tool here: https://react-native-community.github.io/upgrade-helper/?from=0.69.9&to=0.72.4#RnDiffApp-android-gradle-wrapper-gradle-wrapper.properties

**Pull request merged**